lib/go/thrift/server_socket.go

```diff
@@ -110,7 +110,9 @@ func (p *TServerSocket) Open() error {
 }
 
 func (p *TServerSocket) Addr() net.Addr {
-	if p.listener != nil {
+	p.mu.RLock()
+	defer p.mu.RUnlock()
+	if p.IsListening() {
```

Comment:
If you look at the comment about mu at https://github.com/apache/thrift/blob/d9a97c1610372eeb6db9f1a54e590a0a89067aea/lib/go/thrift/server_socket.go#L33-L34, it's only supposed to protect `interrupted` field.

I guess that has evolved over the years, and we already have existing code to use `mu` to also protect `listener`.

so this is the time we should update the comment to reflect that. please move the `listener` field to the same group with `mu` and `interrupted`, and also update the comment.